src/Core/Enums/RuleId.cs

```diff
@@ -40,15 +37,11 @@ public enum RuleId
         ButtonInvokeAndTogglePatterns, // Button should not have Invoke and Toggle patterns together.
         ButtonInvokeAndExpandCollapsePatterns, // Button may have Invoke and ExpandCollapse patterns together. (warning)
         ButtonToggleAndExpandCollapsePatterns, // Button should have have Toggle and ExpandCollapse patterns together.
-        ButtonExpandCollapsePattern,   // obsolete but don't remove since removal will break backward compatibility.
-        ButtonPatterns, // obsolete. keep it to avoid backward compat issue.
```

Comment:
Could this pull request also remove: `src\RulesTest\Library\ButtonPatterns.cs`?

Reply:
Long ago, the `ButtonPatterns` rule was split into multiple pieces. The test file should be renamed to `ButtonShouldHavePatterns`, but it's still needed. I've made a note to make a holistic pass through the `RulesTest` project--some tests append "Tests" to the rule name, while others just use the rule name for the test file.